cloudUtils/getRootCodenamesOfItem.ts

```diff
@@ -61,27 +59,23 @@ const processItem = (itemToProcess: ContentItem, rootItemTypes: string[], contex
 };
 
 const getDirectParents = (codename: string, allItems: ContentItem[]): ContentItem[] =>
-    allItems.filter((item) =>
-        isInAnyRichTextElement(item, codename) ||
-        isInAnyLinkedItemsElement(item.elements as ContentItem, codename));
-
-const isInAnyRichTextElement = (parentItem: ContentItem, codename: string): boolean =>
-    isInElements(parentItem, codename, FieldType.RichText);
-
-const isInAnyLinkedItemsElement = (itemElements: ContentItem, codename: string): boolean =>
-    itemElements && isInElements(itemElements as ContentItem, codename, FieldType.ModularContent);
+    allItems.filter((item) => isInAnyElement(item, codename));
 
-const isInElements = (parentItem: ContentItem, codename: string, fieldType: FieldType): boolean =>
+const isInAnyElement = (
+    parentItem: ContentItem,
+    codename: string): boolean =>
     Object
         .keys(parentItem)
         .map((key) => {
             const element = parentItem[key];
-            if (element.type && element.type === fieldType) {
-                const itemsInElement = element.linkedItemCodenames
-                    ? element.linkedItemCodenames
-                    : element.value;
+            if (element instanceof Elements.RichTextElement) {
+                return element.linkedItemCodenames.includes(codename);
+            }
 
-                return itemsInElement.includes(codename);
+            if (element instanceof Elements.LinkedItemsElement) {
+                return element.itemCodenames.includes(codename);
             }
+
+            return false;
         })
-        .reduce((accumulator, current) => accumulator || current, false);
+        .reduce((accumulator, current) => current === true ? true : accumulator, false);
```

Comment:
I think this could stay as it was, couldn't it?

Reply:
Yeah, it could, I just like the explicit equality check to make it a bit more readable :)